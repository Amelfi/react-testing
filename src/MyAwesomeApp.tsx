import type { CSSProperties } from "react";

const firstName = 'Adan';
const lastName = 'Santana';
const isActive = true;
const arrayData = ['Mario', 'Luffy', 'Warrio'];
const objectData = {
    key: '123ABC',
    articleName: 'Carro'
}

const styleData: CSSProperties = {
    backgroundColor: 'red',
    borderRadius: '10px',
}

export function MyAwesomeApp(){

    return(
        <div>
            <h1 data-testid="first-name-title">{firstName}</h1>
            <h2 style={styleData}>{lastName}</h2>
            <p>{isActive ? 'Activo': 'Inactivo'}</p>
            <p className="mi-clase-con-juegos">{arrayData.join(', ')}</p>
            <p>{JSON.stringify(objectData)}</p>
        </div>
    )
}