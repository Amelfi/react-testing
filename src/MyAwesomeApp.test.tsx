import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";
import {render, screen} from '@testing-library/react'
describe('MyAwesomeApp', ()=>{
    test('Should render firstName and lastName',() => {

        //! 1 Arrange
        //Usar container con el render cuando es estatico.
        //con el container puedo obtener los elementos del componente.
        const {container} = render(<MyAwesomeApp/>);
        //! 2 Act
        const h1 = container.querySelector('h1');
        const h2 = container.querySelector('h2');
        //! 3 Assert
        expect(h1?.innerHTML).toBe('Adan');
        expect(h2?.innerHTML).toBe('Santana');
    });

    test('Should render firtName and lastName screen', () => {

        //! 1 Arrange
        render(<MyAwesomeApp/>);
        //Usar screen cuando hay eventos
        //screen.debug  muestra los elementos del componente.
        // screen.debug();

        //! 2 Act
        // const h1 = screen.getByRole('heading',{
        //     level: 1
        // });

        const h1 = screen.getByTestId('first-name-title');


        //! 3 Assert
        expect(h1.innerHTML).toContain('Adan');
        // screen.getAllByTestId()
    });

    test('Should match snapshot', ()=> {
        const {container} = render(<MyAwesomeApp/>);
        const domHtml = container.innerHTML;
        console.log(domHtml);

        expect(container).toMatchSnapshot();
    });
});