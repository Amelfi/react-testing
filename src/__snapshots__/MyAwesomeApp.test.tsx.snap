// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > Should match snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="first-name-title"
    >
      Adan
    </h1>
    <h2
      style="background-color: red; border-radius: 10px;"
    >
      Santana
    </h2>
    <p>
      Activo
    </p>
    <p
      class="mi-clase-con-juegos"
    >
      Mario, Luffy, Warrio
    </p>
    <p>
      {"key":"123ABC","articleName":"Carro"}
    </p>
  </div>
</div>
`;
